### PR TITLE
cmd/corectl: expose cored error messages

### LINKED ISSUE #1078: corectl does not provide useful error messages
Running an invalid corectl command, such as `corectl create-token foobar invalidpolicy` yields a generic error message (`400 Bad Request`). The error message and detail contained in the body of the response should be relayed to user.

### PR DESCRIPTION
**Pull request by jeffomatic**
This commit allows corectl to report formatted error data (chain code,
message, and detail) to the command-line user.

Fixes #1078

### CODE REVIEW

**Comment by jeffomatic** [comment]
Example output:

```
$ corectl create-token tok-1 foooobar
tok-1:58dd068d30385a0e4afce07242a5214c8d2568421953a591e9daf9affde133bd
Auth grant error:
RPC error: CH003 Invalid request body
Detail: invalid policy: foooobar
$ corectl create-token tok-1 foooobar
RPC error: CH302 Access token id is already in use
Detail: id "tok-1" already in use
```

**Comment by jeffomatic** [comment]
PTAL

**Comment by jeffomatic** [comment]
This will help debugging some of the corectl flows described in our (possibly stale) documentation.

**Comment by kr** [comment]
:eyes:

File: core/rpc/rpc.go [hunk 1]
```diff
@@ -138,11 +140,20 @@ func (c *Client) CallRaw(ctx context.Context, path string, request interface{})
...
 			StatusCode: resp.StatusCode,
 		}
+
+		// Attach formatted error message, if available
+		var errData httperror.Response
+		err := json.NewDecoder(resp.Body).Decode(&errData)
+		if err == nil && errData.ChainCode != "" {
+			resErr.ErrorData = &errData
+		}
+		resp.Body.Close()
```
Comment (kr) [inline core/rpc/rpc.go]: This is unlikely to matter but can you put this call back at the top of the if body in a defer? That way an inadvertent panic in between  there and here won't leak a connection.

**Comment by kr** [comment]
LGTM with a small suggestion

**Pull request merged**